pallets/market/src/benchmarking.rs

```diff
@@ -0,0 +1,409 @@
+        // as such, it should be dropped ASAP so that when a transfer occurs the issuance is "valid"
+        // otherwise, an underflow occurs during the transfer
+        // https://github.com/paritytech/polkadot-sdk/blob/721f6d97613b0ece9c8414e8ec8ba31d2f67d40c/substrate/frame/balances/src/impl_currency.rs#L328-L345
+        // https://github.com/paritytech/polkadot-sdk/blob/721f6d97613b0ece9c8414e8ec8ba31d2f67d40c/substrate/frame/balances/src/impl_currency.rs#L222-L231
+        // https://github.com/paritytech/polkadot-sdk/blob/6eca7647dc99dd0e78aacb740ba931e99e6ba71f/substrate/frame/support/src/traits/tokens/fungible/regular.rs#L317-L339
+        // https://github.com/paritytech/polkadot-sdk/blob/721f6d97613b0ece9c8414e8ec8ba31d2f67d40c/substrate/frame/balances/src/impl_fungible.rs#L104-L151
+        pallet_balances::Pallet::<T>::make_free_balance_be(
+            &caller,
+            // Must add more than the amount that will be added to the market balance
+            // otherwise the account may not have enough to pay fees
+            (EXISTENTIAL_DEPOSIT * 2).into(),
+        );
+
+        // #[extrinsic_call] requires type shenanigans, using #[block] is MUCH simpler
+        #[block]
+        {
+            Pallet::<T>::add_balance(
+                RawOrigin::Signed(caller.clone()).into(),
+                EXISTENTIAL_DEPOSIT.into(),
+            )
+            .unwrap();
+        }
+
+        let balance_entry = BalanceTable::<T>::get(&caller);
+        assert_eq!(balance_entry.free, EXISTENTIAL_DEPOSIT.into());
+        assert_eq!(balance_entry.locked, 0u32.into());
+    }
+
+    #[benchmark]
+    fn withdraw_balance() {
+        let caller: T::AccountId = whitelisted_caller();
+        pallet_balances::Pallet::<T>::make_free_balance_be(
+            &caller,
+            (EXISTENTIAL_DEPOSIT * 2).into(),
+        );
+        // Add some balance so we can withdraw it
+        Pallet::<T>::add_balance(
+            RawOrigin::Signed(caller.clone().into()).into(),
+            EXISTENTIAL_DEPOSIT.into(),
+        )
+        .unwrap();
+
+        // #[extrinsic_call] requires type shenanigans, using #[block] is MUCH simpler
+        #[block]
+        {
+            Pallet::<T>::withdraw_balance(
+                RawOrigin::Signed(caller.clone()).into(),
+                EXISTENTIAL_DEPOSIT.into(),
+            )
+            .unwrap();
+        }
+
+        let balance_entry = BalanceTable::<T>::get(&caller);
+        assert_eq!(balance_entry.free, 0u32.into());
+        assert_eq!(balance_entry.locked, 0u32.into());
+    }
+
+    // When running `cargo test` the first get's called,
+    // for the proper benchmarks we get the second
+    #[cfg(test)]
+    const MAX_DEALS: u32 = 32;
+    #[cfg(not(test))]
+    const MAX_DEALS: u32 = 128;
+
+    /// `n`: number of submitted deals
+    #[benchmark]
+    fn publish_storage_deals(n: Linear<1, MAX_DEALS>) {
```

Comment:
Does the generated benchmark weight is set dynamically based on the number of deals passed to the extrinsic in the runtime?

Reply:
Yes